### PR TITLE
Convert stuff to DSL2

### PR DESCRIPTION
**Pull request by Emiller88**
This definitely doesn't work yet, but no reason to chase our tail until we get some test data.

Much easier to get a lay of the workflow though now.

### CODE REVIEW

File: modules/local/bcl2fastq.nf [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        --runfolder-dir ${runDir} \\
+        --output-dir . \\
+        --sample-sheet ${std_samplesheet} \\
+        --adapter-stringency ${params.adapter_stringency} \\
+        $args \\
+        --minimum-trimmed-read-length ${params.minimum_trimmed_readlength} \\
+        --mask-short-adapter-reads ${params.mask_short_adapter_reads} \\
+        --fastq-compression-level ${params.fastq_compression_level} \\
+        --barcode-mismatches ${params.barcode_mismatches} \\
+        $args2
```
Comment (matthdsm) [inline modules/local/bcl2fastq.nf:41]: Perhaps move this to args and only hard code in and output args

File: modules/local/bcl2fastq.nf [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        --runfolder-dir ${runDir} \\
+        --output-dir . \\
+        --sample-sheet ${sheet} \\
+        --adapter-stringency ${params.adapter_stringency} \\
+        $args \\
+        --minimum-trimmed-read-length ${params.minimum_trimmed_readlength} \\
+        --mask-short-adapter-reads ${params.mask_short_adapter_reads} \\
+        --fastq-compression-level ${params.fastq_compression_level} \\
+        --barcode-mismatches ${params.barcode_mismatches}
+        $args2
```
Comment (matthdsm) [inline modules/local/bcl2fastq.nf:57]: Perhaps move this to args and only hard code in and output args.
Also, what's the meaning of arg1 and 2 when there's only one executable?

**Review by matthdsm — CHANGES_REQUESTED** [review]
Perhaps it would be best to drop the QC steps and focus on the demultiplexing aspect for a first version.
So that'd mean dropping cellranger count, kraken, and fastq_screen.

File: modules/local/bcl2fastq.nf [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        bcl2fastq \\
+        --runfolder-dir ${runDir} \\
+        --output-dir . \\
+        --sample-sheet ${sheet} \\
+        --adapter-stringency ${params.adapter_stringency} \\
+        $args \\
+        --minimum-trimmed-read-length ${params.minimum_trimmed_readlength} \\
+        --mask-short-adapter-reads ${params.mask_short_adapter_reads} \\
+        --fastq-compression-level ${params.fastq_compression_level} \\
+        --barcode-mismatches ${params.barcode_mismatches}
```
Comment (matthdsm) [inline modules/local/bcl2fastq.nf:56]: this is missing a `\\`

**Comment by Emiller88** [comment]
> Perhaps it would be best to drop the QC steps and focus on the demultiplexing aspect for a first version.
So that'd mean dropping cellranger count, kraken, and fastq_screen.

100%, I just wanted to split these out, so we could get a cleaner picture of what v1 looked like. Do you think we should even support bcl2fastq?

File: modules/local/bcl2fastq.nf (hunk 2 above)
Comment (Emiller88) [reply modules/local/bcl2fastq.nf]: I just split those up because there were so many args they wanted to handle.

**Comment by matthdsm** [comment]
I'd go with `bcl-convert` and `bases2fastq` as a starting point. We can expand to `bcl2fastq`, `picard` and others in later versions as needed

**Comment by matthdsm** [comment]
Feel free!

**Pull request merged**